### DISCUSSION TITLE
Adding a command to TUTD for showing the native expression sent to the server

### DISCUSSION THREAD

**Discussion by farzadbekran** [agentm/project-m36#308] [Ideas]
TUTD is clearly easier to work with. Making the equivalent expressions with the native data types is really hard sometimes, especially for constraints. It would be nice if we had a flag that enabled logging the resulting expressions that tutd interprets, in native format.

**Comment by agentm** [comment]
We do have a ticket to implement [template haskell support for the TutorialD language](#288). With that, you would be able to get the underlying ADT from GHCi. I suppose that could be used as a short-term solution for this until the server has proper logging.
